### PR TITLE
[MOB-486] Possible fix for Wallet validation problem

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

   Tries to fix issue that users are facing with WalletValidation.
   Added logs to better understand problem

**Database changed?**

   Yes | No

**Where should the reviewer start?**

PhoneValidationPresenter.kt

**How should this be manually tested?**

Check if it didn't break anythin

**What are the relevant tickets?**

 https://aptoide.atlassian.net/browse/MOB-486

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/DebugLogger.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    if (BuildConfig.DEBUG) {
+      throwable.printStackTrace()
+      Log.d("Logger", throwable.message)
+    }
+  }
+
+  override fun log(message: String) {
+    val throwable = Throwable(message)
+    log(throwable)
+  }
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/DebugLogger.kt]: no need to declare throwable, just create it inside log ?

File: app/src/main/java/com/asfoundation/wallet/Logger.java [hunk 2]
```diff
@@ -3,4 +3,6 @@
 public interface Logger {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/Logger.java:3]: what is this extension? .java?

image

File: app/src/main/java/com/asfoundation/wallet/FlurryLogger.java [hunk 3]
```diff
@@ -11,4 +11,9 @@
       FlurryAgent.onError("ID", throwable.getMessage(), throwable);
     }
   }
+
+  @Override public void log(String message) {
+    Throwable throwable = new Throwable(message);
+    log(throwable);
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/FlurryLogger.java]: no need to declare throwable, just create it inside log ?

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/Logger.java (hunk 2 above)
Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/Logger.java:3]: I'll leave that change for the future ticket :p

**Review by carvalho2707 — APPROVED** [review]

**Pull request merged**